src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11ECKeyFactory.java

```diff
@@ -294,7 +294,13 @@ private PrivateKey generatePrivate(BigInteger s, ECParameterSpec params)
             try {
                 token.p11.C_GetAttributeValue(session[0].id(), keyID, attributes);
                 ECParameterSpec params = decodeParameters(attributes[1].getByteArray());
-                ECPoint point = decodePoint(attributes[0].getByteArray(), params.getCurve());
+                ECPoint point;
+
+                if (!token.config.getUseEcX963Encoding()) {
+                    point = decodePoint(new DerValue(attributes[0].getByteArray()).getOctetString(), params.getCurve());
+                } else {
+                    point = decodePoint(attributes[0].getByteArray(), params.getCurve());
+                }
```

Comment:
You could save a "!" operation in 299 if switch line 302 and line 300.

Reply:
I only used "!" for consistency with existing usage in PK11Key.java:1080.    Is there a reason to avoid "!" other than maybe readability?

Reply:
Save a operation could get a little bit performance.  Comparing to "if (!a)", "if (a)" is easier to read to me, and save me a cycle to compute the "!".  Anyway, not a big concern of mine, you can leave it as is if you prefer the "if (!a)" style.

Reply:
Just to finish up the thought.  The assembly instructions are the same using if(!a) or if(a).  The difference would be how the code is setup during compile time and I believe that would be the same speed as well..

Reply:
Thank you for the check.  It looks good to me and I have no more comment.